crates/uv-distribution-filename/src/wheel.rs

```diff
@@ -202,25 +216,58 @@ impl WheelFilename {
                     .map_err(|err| WheelFilenameError::InvalidBuildTag(filename.to_string(), err))
             })
             .transpose()?;
+
+        let tags = if build_tag.is_some()
+            || python_tag.contains('.')
+            || abi_tag.contains('.')
+            || platform_tag.contains('.')
```

Comment:
This is a bit wasteful -- we should instead use an iterator, call `.next()`, and then check if the _next_ `.next()` is `None` for all of them. But that's a lot more code, unclear if it's really worth it.

Reply:
Interesting. I would naively assume that it would matter, since it will be running for what looks like every wheel filename parse? Or is there some logic here that results in a short circuit in most cases?

Reply:
Well, it doesn't matter for _small_ tags, right? Since we have to check this regardless? But it does matter for _large_ tags, which is already "expensive".

Reply:
Yeaaaaah, okay, I see what you're saying now. I agree that this is fine.

I think the only possible improvement here would be to bake this conditional logic into the parsing of tags itself. Something like, "parse the first tag and tell me if it looks like there are more tags to parse" _without_ first splitting on `.`. But I'm not even sure that would be a win, and it makes the parsing more complicated.